bookkeeper-server/src/main/java/org/apache/bookkeeper/util/collections/ConcurrentLongLongPairHashMap.java

```diff
@@ -376,7 +440,16 @@ private boolean remove(long key1, long key2, long value1, long value2, int keyHa
                 }
 
             } finally {
-                unlockWrite(stamp);
+                if (autoShrink && size < resizeThresholdBelow) {
+                    try {
+                        // shrink the hashmap
+                        rehash((int) (capacity * (1 - shrinkFactor)));
```

Comment:
Shouldn't this be like:

```suggestion
                        rehash((int) (capacity / shrinkFactor));
```

Reply:
I agree.
In addition, should the parameter range check be added? like: @merlimat 
shrinkFactor>1
expandFactor>1

Reply:
👍